hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/clustering/ClusteringPlanSourceFunction.java

```diff
@@ -60,21 +61,31 @@ public class ClusteringPlanSourceFunction extends AbstractRichFunction implement
    */
   private final String clusteringInstantTime;
 
-  public ClusteringPlanSourceFunction(String clusteringInstantTime, HoodieClusteringPlan clusteringPlan) {
+  private boolean isPending;
+
+  private final Configuration conf;
+
+  public ClusteringPlanSourceFunction(String clusteringInstantTime, HoodieClusteringPlan clusteringPlan, Configuration conf) {
     this.clusteringInstantTime = clusteringInstantTime;
     this.clusteringPlan = clusteringPlan;
+    this.conf = conf;
   }
 
   @Override
   public void open(Configuration parameters) throws Exception {
-    // no operation
+    isPending = StreamerUtil.createMetaClient(conf).getActiveTimeline()
+        .getInstantsAsStream().anyMatch(i -> clusteringInstantTime.equals(i.getTimestamp()) && !i.isCompleted());
```

Comment:
There is a method named `containsInstant`, can we use that?

Reply:
Maybe we can also move the code into `run` so there is no need to keep a class member `isPending`.